gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -261,53 +262,127 @@ fn read_auxi_check_from_tlvc(data: &[u8]) -> Result<[u8; 32], UpdateError> {
+                    // The Bootleby bootloader has previously been packaged
+                    // as a simple zip archive without the extra information
+                    // found in the Hubris-style zip archive. The old format
+                    // has been used in manufacturing and for updating machines
+                    // with debug probes attached.
+                    //
+                    // When we are satisfied with automated update of bootleby,
+                    // then updated manufacturing images and rollback
+                    // protection (to be implemented) will allow us to remove
+                    // support for the old image format. Until then, we need
+                    // to be able to update and rollback using the old and new
+                    // releases.
+                    //
+                    // So, for now, access the Bootleby archive as a plain zip
+                    // file where we are looking for either 'bootleby.bin' or
+                    // 'img/final.bin'. Later, use RawHubrisArchive as above.
+                    let contents = image.clone();
+                    let cursor = Cursor::new(contents.as_slice());
+                    let mut archive = match zip::ZipArchive::new(cursor) {
+                        Ok(archive) => archive,
+                        Err(_) => return Err(UpdateError::InvalidArchive),
+                    };
+
+                    // Support old format archives for now.
+                    let mut rot_image = vec![];
+                    for i in 0..archive.len() {
+                        let mut file = match archive.by_index(i) {
+                            Ok(file) => file,
+                            Err(e) => {
+                                error!(
+                                    log,
+                                    "did not find bootloader file: {}", e
+                                );
+                                return Err(UpdateError::InvalidArchive);
+                            }
+                        };
+                        if matches!(
+                            &file.name(),
+                            &"bootleby.bin" | &"img/final.bin"
+                        ) {
+                            if file.read_to_end(&mut rot_image).is_err() {
+                                error!(log, "invalid archive");
+                                return Err(UpdateError::InvalidArchive);
+                            }
+                            debug!(
+                                log,
+                                "found bootloader file {}",
+                                &file.name()
+                            );
+                            break;
+                        }
+                    }
+                    rot_image
+                }
+                _ => return Err(UpdateError::InvalidSlotIdForOperation),
+            }
+        }
+        _ => return Err(UpdateError::InvalidComponent),
+    };
```

Comment:
Can we follow a similar pattern to what we have in embootleby? https://github.com/oxidecomputer/embootleby/blob/main/src/main.rs#L193-L239 We try the hubtools format first and if that doesn't work fall back to manually reading the zip archive. I think that will be cleaner to remove in the future.

Reply:
I made a change to that effect. Thanks!